src/run.rs

```diff
@@ -240,8 +268,14 @@ impl Handler for ExecHandler {
             // Default behaviour (neither --signal nor --restart specified):
             // Make sure the previous run was ended, then run the command again
             (false, false) => {
-                signal_process(&self.child_process, None, true);
-                self.spawn(ops)?;
+                if self.args.watch_when_idle {
+                    if !running_process {
+                        self.spawn(ops)?;
+                    }
+                } else {
+                    signal_process(&self.child_process, None, true);
+                    self.spawn(ops)?;
+                }
```

Comment:
Can you try instead to add a third boolean to the match and see if that works out better? having the same if three times feels a bit weird

So like 

```rust
let scenario = (self.args.restart, self.signal.is_some(), self.args.watch_when_idle);
match scenario {
    // SIGHUP scenario (needs to be up here to take precedence)
    (false, true, _) => signal_process(&self.child_process, self.signal, false),
    // watch-when-idle
    (_, _, true) => if !running_process { self.spawn(ops)?; },
    // etc
    (true, true, false) => {
        signal_process(&self.child_process, self.signal, true);
        self.spawn(ops)?;
    },
    // ...
}
```

Reply:
Yeah, this makes sense..